Group together extended reference doc sections (#9908)
apollographql/router: docs/source/routing/configuration/yaml.mdx

}
```

#### Extended error reporting

<MinVersionBadge version="Router v2.1.2" />

<PreviewFeature />

<Note>

The router supports extended error reporting in the following versions:

- [Preview](/resources/product-launch-stages/#preview) in v2.1.2 and later
- [Experimental](/resources/product-launch-stages/#experimental-features) in v2.0.0

</Note>

You can configure the router to report extended error information for improved diagnostics.
Apollo's legacy error reporting doesn't include the service or error code, meaning you can't easily attribute errors to their root cause in GraphOS Studio.

> Configure extended reference reporting in `router.yaml` with the `telemetry.apollo.errors.preview_extended_error_metrics` option like so:

Reviewer: ```suggestion
Configure extended reference reporting in `router.yaml` with the `telemetry.apollo.errors.preview_extended_error_metrics` option:
```

```yaml title="router.yaml"
telemetry:
  apollo:
    errors:
      preview_extended_error_metrics: enabled # Default is disabled
```

> [Learn more.](/graphos/routing/graphos-reporting#errors)

Reviewer: ```suggestion
[Learn more about GraphOS extended error reporting](/graphos/routing/observability/graphos/graphos-reporting#enabling-extended-error-reporting).
```

#### Extended reference reporting

<MinVersionBadge version="Router v1.50.0" />

<Note>

The router supports extended reference reporting in the following versions:

- [General availability](/resources/product-launch-stages/#general-availability) in v1.54.0 and later
- [Experimental](/resources/product-launch-stages/#experimental-features) in v1.50.0 to v1.53.0

</Note>

You can configure the router to report enum and input object references for enhanced insights and operation checks.
Apollo's standard reference reporting doesn't include data about enum values and input object fields, meaning you can't view enum and input object field usage in GraphOS Studio.
Standard reporting can also cause [inaccurate operation checks](#enhanced-operation-checks).

> To configure reference reporting, update `telemetry.apollo.metrics_reference_mode` in `router.yaml`:

Reviewer: ```suggestion
Configure reference reporting by updating `telemetry.apollo.metrics_reference_mode` in `router.yaml`:
```
